CMakeLists.txt

```diff
@@ -19,8 +22,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-find_package(MPI REQUIRED)
-find_package(Threads REQUIRED)
+find_package(MPI 3.1 REQUIRED C)
```

Comment:
Should probably be 
```
find_package(MPI 3.1 REQUIRED CXX)
```
See https://cmake.org/cmake/help/latest/module/FindMPI.html

Reply:
Just for a link, @TLCFEM noticed that 
```
find_package(MPI 3.1 REQUIRED CXX)
```
should be enough since we are using C++. https://github.com/rabauke/mpl/commit/80c3305d90e2e3927cee7a8710fd93606d2c8667#r158914527